src/escrows/ConvexEscrow.sol

```diff
@@ -0,0 +1,142 @@
+        _; 
+    }
+
+    modifier onlyBeneficiaryOrAllowlist {
+        require(msg.sender == beneficiary || allowlist[msg.sender], "ONLY BENEFICIARY OR ALLOWED");
+        _; 
+    }
+
+    event AllowClaim(address indexed allowedAddress, bool allowed);
+
+    /**
+     * @notice Initialize escrow with a token
+     * @dev Must be called right after proxy is created.
+     * @param _beneficiary The beneficiary who cvxCRV is staked on behalf
+     */
+    function initialize(address, address _beneficiary) public {
+        require(market == address(0), "ALREADY INITIALIZED");
+        market = msg.sender;
+        token.approve(address(rewardPool), type(uint).max);
+        beneficiary = _beneficiary;
+    }
+
+    /**
+     * @notice Withdraws the wrapped token from the reward pool and transfers the associated ERC20 token to a recipient.
+     * @param recipient The address to receive payment from the escrow
+     * @param amount The amount of ERC20 token to be transferred.
+     */
+    function pay(address recipient, uint amount) public {
+        require(msg.sender == market, "ONLY MARKET");
+        stakedBalance -= amount;
+        rewardPool.withdraw(amount, false);
+        token.transfer(recipient, amount);
+    }
+
+    /**
+     * @notice Get the token balance of the escrow
+     * @return Uint representing the staked balance of the escrow
+     */
+    function balance() public view returns (uint) {
+        return stakedBalance;
+    }
+
+    /**
+     * @notice Function called by market on deposit. Stakes deposited collateral into convex reward pool
+     * @dev This function should remain callable by anyone to handle direct inbound transfers.
+     */
+    function onDeposit() public {
+        //Stake cvxCRV
+        uint tokenBal = token.balanceOf(address(this));
+        stakedBalance += tokenBal;
+        rewardPool.stake(tokenBal);
+    }
+
+    /**
+     * @notice Claims reward tokens to the specified address. Only callable by beneficiary and allowlisted addresses
+     * @param to Address to send claimed rewards to
+     */
+    function claimTo(address to) public onlyBeneficiaryOrAllowlist {
+
+        uint extraRewards = rewardPool.extraRewardsLength();
+        //Claim rewards, only claim extra rewards if there are any
+        rewardPool.getReward(address(this), extraRewards > 0, false);
+        rewardToken.transfer(to, rewardToken.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Claims reward tokens to the specified address. Only callable by beneficiary and allowlisted addresses
+     * @param to Address to send claimed rewards to
+     * @param extraRewards Array of token addresses of extra rewards that will be sent to `to`
+     */
+    function claimTo(address to, address[] calldata extraRewards) external onlyBeneficiaryOrAllowlist {
+        require(rewardPool.extraRewardsLength() == extraRewards.length, "UNEQUAL ARRAY");
+        claimTo(to);
+        for(uint i; i < extraRewards.length; ++i){
```

Comment:
no validation ensuring user-provided extraRewards addresses actually match the extraRewards addresses in Convex contract.

but given you've already blocked collateral claims, i take it there's no other unsafe tokens expected to in this contract. i can't think of any myself.